snap/info_snap_yaml.go

```diff
@@ -440,16 +444,26 @@ func validateAttr(v interface{}) (interface{}, error) {
 		for k, item := range x {
 			kStr, ok := k.(string)
 			if !ok {
-				return nil, fmt.Errorf("non-string key in attribute map: %v", k)
+				return nil, fmt.Errorf("non-string key in %s map: %v", errorContext, k)
```

Comment:
Ok, please have a look now, I hope I got the idea right. I've removed the context argument and tweaked existing messages to be more generic so that normalizeYamlValue() function is usable both from attribute validation code and gadget defaults validation.